api/codes.py

```diff
@@ -0,0 +1,45 @@
+import json
+import requests
+import country_converter as coco
+
+cc = coco.CountryConverter()
+
+cities = open('cities.json')
+city_data = json.load(cities)
+
+airports = open('airports.json')
+airport_data = json.load(airports)
+
+
+def convert_country_code(city, country):
+    country_code = cc.convert(names=country, to='ISO2')
+    return convert_city(city, country_code)
+
+
+def convert_city(city, country):
+    city_code = str()
+    for x in city_data:
+        if x['name_translations']['en'] == city and x['country_code'] == country:
+            # print(x['name_translations']['en'])
+            # city_code.append(x['name_translations']["en"])
+            city_code = x['code']
```

Comment:
Personally, I think its better to write declarative code over imperative code, since it is overall neater and more readable. The following code here can be expressed as 
```
city_index = index(city_data, lambda x: x['name_translations']['en'] == city and x['country_code'] == country)
city_code = city_data[city_index]['code]
```

Reply:
I feel imperative is better for me as it is easier to understand and goes through step by step on how the code will work.

Reply:
Ok, atleast put a `break` so that it ends the loop after it finds

Reply:
Also whut happens if the city is not found?

Reply:
What package did the index method come from?

Reply:
> Also whut happens if the city is not found?

The function will return an empty variable. I will need to cater for that before moving it onto the next function.

Reply:
> 

Sry I'm high, supposed to be `city_data.index(...)`

Reply:
> > 
> 
> Sry I'm high, supposed to be `city_data.index(...)`

i don't think lambda works with index... just tried it and it throws exceptions. Ill see if there are any other functions or methods i can use.

Reply:
Ah, try
```
def find(l, f):
     return next((l[i] for i in xrange(len(l)) if f(l[i])), None)

city = find(city_data, lambda x: x['name_translations']['en'] == city and x['country_code'] == country)
city_code = city['code']
```

Reply:
The `find` function can be put into a seperate utils file for use else where. Kinda odd tho, expected python to have a built in list find function

Reply:
> Ah, try
> 
> ```
> def find(l, f):
>      return next((l[i] for i in xrange(len(l)) if f(l[i])), None)
> 
> city = find(city_data, lambda x: x['name_translations']['en'] == city and x['country_code'] == country)
> city_code = city['code']
> ```

oh i managed to get it to work already using 2 different methods

either:
`city_index = list(filter(
        lambda x: city_data[x]['name_translations']['en'] == city and city_data[x]['country_code'] == country, range(len(city_data))))`

or

`city_index = [city_data.index(x) for x in city_data if x['name_translations']['en'] == city and x['country_code'] == country]`

Reply:
> The `find` function can be put into a seperate utils file for use else where. Kinda odd tho, expected python to have a built in list find function

I would guess the find function would be the filter function for python... not too sure tho hahaha